projects/stargazer/plugins/other/radius/radius.h

```diff
@@ -10,21 +14,26 @@ namespace STG
         public:
             RADIUS();
 
-            int Start() override { return 0; }
-            int Stop() override { return 0; }
+            int Start() override;
+            int Stop() override;
             int Reload(const ModuleSettings & /*ms*/) override { return 0; }
-            bool IsRunning() override { return isRunning; }
+            bool IsRunning() override { return m_running; }
+            void SetRunning(bool val);
             int ParseSettings() override { return 0; }
-            const std::string & GetStrError() const override { return errorStr; }
+            const std::string & GetStrError() const override { return m_errorStr; }
             std::string GetVersion() const override;
             uint16_t GetStartPosition() const override { return 0; }
             uint16_t GetStopPosition() const override { return 0; }
 
             int SendMessage(const Message & msg, uint32_t ip) const override { return 0; }
 
         private:
-            mutable std::string errorStr;
-            bool isRunning;
+            mutable std::string m_errorStr;
+            std::jthread m_thread;
+            std::mutex m_mutex;
```

Comment:
A little pro-tip. Always put the mutex first in the member list. This way you ensure that it is initialized when you use it.

I had an issue in the past when I put the mutex after the thread, like you here. And I started the thread right inside the initialization, and it ran before the mutex was initialized, and inside the thread function I tried to lock the mutex and everything crashed, because the mutex was not initialized. It was a really tricky issue to find, because most time the thread starts with a little delay, and the mutex has a chance to be initialized.